intern/convenience/array.py

```diff
@@ -533,47 +533,53 @@ def __getitem__(self, key: Tuple) -> np.array:
         if self._coord_frame is None:
             self._populate_coord_frame()
 
-        _normalize_units = (1, 1, 1)
-        if isinstance(key[-1], str) and len(key) == 4:
-            if key[-1] != self._coord_frame.voxel_unit:
-                raise NotImplementedError(
-                    "Can only reference voxels in native size format which is "
-                    f"{self._coord_frame.voxel_unit} for this dataset."
-                )
-            _normalize_units = self.voxel_size
-
-        if isinstance(key[2], int):
-            xs = (key[2], key[2] + 1)
+        if isinstance(key, int):
+            # Get the full Z slice:
+            xs = (0, self.shape[2])
+            ys = (0, self.shape[1])
+            zs = (key, key + 1)
         else:
-            start = key[2].start if key[2].start else 0
-            stop = key[2].stop if key[2].stop else self.shape[0]
+            _normalize_units = (1, 1, 1)
+            if isinstance(key[-1], str) and len(key) == 4:
```

Comment:
Can you add comments that describe each indexing case you're handling? That'd make it much easier for the next person who might have to change things.

Reply:
Added!